include/ginkgo/core/log/logger.hpp

```diff
@@ -658,10 +664,44 @@ class EnableLogging : public PolymorphicBase {
 
     void clear_loggers() override { loggers_.clear(); }
 
+private:
+    template <typename OtherLoggable, typename OtherBase>
+    friend class EnableLogging;
```

Comment:
Is that necessary? I could remove it and still build the library.

Reply:
seems like it, I'll remove it